discojs/src/client/client.ts

```
@@ -63,23 +175,13 @@ export abstract class Client extends EventEmitter<{'status': RoundStatus}>{
   }
 
   /**
-   * Communication callback called at the beginning of every training round.
-   */
-  abstract onRoundBeginCommunication(): Promise<void>;
-
-  /**
-   * Communication callback called the end of every training round.
-   * @param weights The local weight update resulting for the current local training round
-   * @returns aggregated weights or the local weights upon error
-   */
-  abstract onRoundEndCommunication(weights: WeightsContainer): Promise<WeightsContainer>;
-
-  // Number of contributors to a collaborative session
-  // If decentralized, it should be the number of peers
-  // If federated, it should the number of participants excluding the server
-  // If local it should be 1
+  * Number of contributors to a collaborative session
+  * If decentralized, it should be the number of peers
+  * If federated, it should the number of participants excluding the server
+  * If local it should be 1
+  */
   get nbOfParticipants(): number {
-    return this.aggregator.nodes.size // overriden by the federated client
+    return 1 // overriden by the federated and decentralized clients
```

would be a good place for `abstract` but abstract accessor doesn't exist in this language.. either move it to a full method `getNbOfParticipants(): number` or at least make it throw a "should be overriden by subclass". better to fail loudly rather than returning an erroneous value.